NebulaPatcher/Patches/Dynamic/EnemyUnitComponent_Patch.cs

```diff
@@ -0,0 +1,400 @@
+﻿#region
+
+using System;
+using HarmonyLib;
+using NebulaWorld;
+using UnityEngine;
+
+#pragma warning disable IDE1006
+#pragma warning disable IDE0007
+#pragma warning disable IDE0018
+
+#endregion
+
+namespace NebulaPatcher.Patches.Dynamic;
+
+[HarmonyPatch(typeof(EnemyUnitComponent))]
+internal class EnemyUnitComponent_Patch
+{
+    [HarmonyPostfix]
+    [HarmonyPatch(nameof(EnemyUnitComponent.GetTargetPosition_Ground))]
+    public static void GetTargetPosition_Ground_Postfix(int idType, ref Vector3 target, ref bool __result)
+    {
+        if (!Multiplayer.IsActive)
+        {
+            return;
+        }
+        var targetType = idType >> 26;
+        var targetId = idType & 67108863;
+        if (targetType == 15) //player
+        {
+            var players = Multiplayer.Session.Combat.Players;
+            for (var i = 0; i < players.Length; i++)
+            {
+                if (players[i].id == targetId)
+                {
+                    target = players[i].position;
+                    __result = players[i].isAlive;
+                    return;
+                }
+            }
+        }
+    }
+
+    [HarmonyPrefix]
+    [HarmonyPatch(nameof(EnemyUnitComponent.GetTargetPosition_Space))]
+    public static bool GetTargetPosition_Space_Postfix(int idType, ref VectorLF3 target, ref float territory, ref bool __result)
+    {
+        if (!Multiplayer.IsActive) return true;
+
+        var targetType = idType >> 26;
+        var targetId = idType & 67108863;
+        if (targetType == 15) //player
+        {
+            if (!Multiplayer.Session.Combat.IndexByPlayerId.TryGetValue(targetId, out var index))
+            {
+                __result = false;
+                return false;
+            }
+
+            ref var ptr = ref Multiplayer.Session.Combat.Players[index];
+            target = ptr.uPosition;
+            territory = GameMain.mainPlayer.mecha.energyShieldRadius; // Currently the shieldRadius are same for all mecha
+            __result = ptr.isAlive;
+            return false;
+        }
+        return true;
+    }
+
+    [HarmonyPostfix]
+    [HarmonyPatch(nameof(EnemyUnitComponent.SensorLogic_Ground))]
+    static void SensorLogic_Ground_Postfix(ref EnemyUnitComponent __instance, ref EnemyData enemy, PlanetFactory factory, int hatred_max_sense, EAggressiveLevel aggressiveLevel)
+    {
+        if (!Multiplayer.IsActive)
+        {
+            return;
+        }
```

Comment:
Minor nitpick, could be simplified to:
```suggestion
        if (!Multiplayer.IsActive) return;
```